File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
    def sumNumbers(self, root: Optional[TreeNode],ans=0) -> int:
        if root==None:
            return 0
        
        ans = ans *10 + root.val
        
        if root.right==None and root.left==None:
            return ans
        
        
        return self.sumNumbers(root.left,ans) + self.sumNumbers(root.right,ans)

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumNumbers(TreeNode* root, int currSum = 0) {
        if (root==NULL) return 0;
        
        currSum = 10* currSum + root->val;
        
        if (root->left==NULL and root->right == NULL) return currSum;
        
        return sumNumbers(root->left, currSum) + sumNumbers(root->right, currSum);
    }
};